### PR TITLE
docs - resource mgmt topic restructure, resource group initial "using" content

### PR DESCRIPTION
**Pull request by lisakowen**
this PR includes a restructure of the Workload Management section of "Administrator Guide -> Managing Performance":
- created a new topic named "Greenplum Database Memory Overview"
- "Workload Management with Resource Queues" topic renamed to "Managing Workload and Resources"; this new topic now has two subordinate topics "Using Resource Queues" and "Using Resource Groups"
  
content updates/additions include:
- "Using Resource Queues" - original "Workload Management with Resource Queues" content with the general memory information info stripped out
- "Greenplum Database Memory Overview" - general memory info that was stripped out of original resource queue discussion
     - added xref to "Managing Workload and Resources" page
	 - misc wording edits
	 - remove "statement_mem" discussion, it is applicable only to resource queues
- "Managing Workload and Resources" - brief intro to workload management and the 2 schemes (resource queues and resource groups) including a comparison
- "Using Resource Groups" - new content describing Resource Groups - Limits, Example, Configuring RGs, Using RGs, Monitoring RG Status

REVIEW CONSIDERATIONS - please review for correctness, not completeness. the resource group feature is still under development. we expect to continue to build on the doc content in the RG area in future PRs - i.e. more examples, documenting additional GUCs and associated discussion, adding other monitoring queries, best practices, etc.

links to updated/new pages on docs review staging site:
- http://docs-gpdb-review-staging.cfapps.io/500Beta/admin_guide/wlmgmt_intro.html - greenplum memory overview
- http://docs-gpdb-review-staging.cfapps.io/500Beta/admin_guide/wlmtmg.html - intro to RG/RQs
- http://docs-gpdb-review-staging.cfapps.io/500Beta/admin_guide/workload_mgmt.html - using resource queues
- http://docs-gpdb-review-staging.cfapps.io/500Beta/admin_guide/workload_mgmt_resgroups.html - using resource groups

### CODE REVIEW

File: gpdb-doc/dita/admin_guide/wlmgmt.xml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  <shortdesc>Greenplum Database workload management helps you prioritize and allocate resources to
+    queries according to business requirements and to prevent queries from starting when resources
+    are unavailable.</shortdesc>
+  <body>
+    <p>Using workload management, you can limit the number of concurrent queries, the amount of memory used to execute a query, and the relative amount of CPU devoted to processing a query. Greenplum Database provides two schemes to manage workload and resources - Resource Queues and Resource Groups.</p>
+      <note>Resource group-based workload management is a Greenplum Database Beta feature currently under development.</note>
+    <p>Either resource queue-based or resource group-based workload management is active in Greenplum Database; both schemes can not be active at the same time.</p>
+    <p>Resource queue-based workload management is enabled by default when you install your Greenplum Database cluster. You can create and assign resource groups while resource queues are active, but you must explicitly enable resource groups to start using that workload management scheme.</p>
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/wlmgmt.xml]: Recommended edit:  ", but you must explicitly enable resource groups to " -> "&lt;/p>&lt;p>You must explicitly enable resource groups to"

File: gpdb-doc/dita/admin_guide/wlmgmt.xml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    <p>While both resource queues and resource groups help you to manage your Greenplum Database workload, the schemes differ in the limits and tools they expose to provide this management.</p>
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/wlmgmt.xml]: Would probably just delete this para.

File: gpdb-doc/dita/admin_guide/wlmgmt.xml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          </thead>
+          <tbody>
+            <row>
+              <entry colname="col1">Concurrency</entry>
+              <entry colname="col2">Managed at the query level</entry>
+              <entry colname="col3">Managed at the transaction level</entry>
+            </row>
+            <row>
+              <entry colname="col1">CPU</entry>
+              <entry colname="col2">Identify by query priority</entry>
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/wlmgmt.xml]: To keep things parallel, maybe change "Identify by" to "Specify".  Also change "Specify percentage" to "Specify percentage of CPU resources"?

File: gpdb-doc/dita/admin_guide/wlmgmt.xml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+              <entry colname="col3">Specify percentage; uses Linux Control Groups</entry>
+            </row>
+            <row>
+              <entry colname="col1">Memory</entry>
+              <entry colname="col2">Managed at the operator level; users can over-subscribe</entry>
+              <entry colname="col3">Enhanced allocation and tracking at the transaction level</entry>
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/wlmgmt.xml]: Change to "Managed at the transaction level, with enhanced..."  Also, seems like we should call out that users can't over-subscribe if that's a difference between the two.

File: gpdb-doc/dita/admin_guide/wlmgmt.xml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            </row>
+            <row>
+              <entry colname="col1">Users</entry>
+              <entry colname="col2">Limits are applied only to non-admin users</entry>
+              <entry colname="col3">Limits are applied to <codeph>SUPERUSER</codeph> and non-admin users alike</entry>
+            </row>
+            <row>
+              <entry colname="col1">Queueing</entry>
+              <entry colname="col2">Queue only when no slot available; query may fail immediately if not enough memory</entry>
+              <entry colname="col3">Queue when no slot is available or not enough available memory</entry>
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/wlmgmt.xml]: Call out whether RG scheme may cause queries to fail or not.

File: gpdb-doc/dita/admin_guide/workload_mgmt_resgroups.xml [hunk 6]
```diff
@@ -0,0 +1,466 @@
...
+        <p>
+          <codeblock>=# CREATE RESOURCE GROUP <i>rgroup1</i> WITH (CPU_RATE_LIMIT=20, MEMORY_LIMIT=25);
+</codeblock>
+        </p>
+        <p>The CPU limit of 20 is shared by every role to which <codeph>rgroup1</codeph> is assigned. Similarly, the memory limit of 25 is shared by every role to which <codeph>rgroup1</codeph> is assigned. <codeph>rgroup1</codeph> utilizes the default <codeph>CONCURRENCY</codeph> setting of 20.</p>
+    </body>
+  </topic>
+
+  <topic id="topic17" xml:lang="en">
+    <title id="iz172210">Assigning a Resource Group to a Role</title>
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/workload_mgmt_resgroups.xml]: I think info about he default groups shoulf come earlier, probably just after the info on enabling RG-based management.

**Comment by lisakowen** [comment]
@dyozie - incorporated your requested edits.

**Comment by lisakowen** [comment]
@gaos1 - can you or your team review the "using resource groups" content?  if you are not interested in the restructuring, you can focus on the two pages below.

REVIEW CONSIDERATIONS - please review for correctness, not completeness. the resource group feature is still under development. we expect to continue to build on the doc content in the RG area in future PRs - i.e. more examples, documenting additional GUCs and associated discussion, adding other monitoring queries, best practices, etc.

nicely formatted links:
- http://docs-gpdb-review-staging.cfapps.io/500Beta/admin_guide/wlmtmg.html - intro to RG/RQs
- http://docs-gpdb-review-staging.cfapps.io/500Beta/admin_guide/workload_mgmt_resgroups.html - using resource groups


thank you.

**Comment by dyozie** [comment]
@gaos1 - Just reiterating Lisa's call for tech review here.  The RG docs will make their debut with the upcoming 5.0.0 GA release.  And even though the feature itself isn't quite GA we'd like to make sure it's as accurate as possible.  Thanks!

**Comment by gaos1** [comment]
I am reviewing Lisa's updates. when all of resource group docs are done,
please let us know as well. we can have another round of reviews before GA.

File: gpdb-doc/dita/admin_guide/workload_mgmt_resgroups.xml [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+      </table>
+
+  </body>
+
+  <topic id="topic8339717179" xml:lang="en">
+    <title>Transaction Concurrency Limit</title>
+    <body>
+      <p>The <codeph>CONCURRENCY</codeph> limit controls the maximum number of concurrent transactions permitted for the resource group. Each resource group is logically divided into a fixed number of slots equal to the <codeph>CONCURRENCY</codeph> limit. Greenplum Database allocates these slots an equal, fixed percentage of memory resources.</p>
+      <p>The default <codeph>CONCURRENCY</codeph> limit value for a resource group is 20.</p>
+      <p>Greenplum Database queues any transactions submitted after the resource group reaches its <codeph>CONCURRENCY</codeph> limit. When a running transaction completes, Greenplum Database un-queues and executes the earliest queued transaction if sufficient CPU and memory resources exist.</p>
```
Comment (gaos1) [inline gpdb-doc/dita/admin_guide/workload_mgmt_resgroups.xml]: only memory resource can become a block issue to un-queue a queued transaction. cpu won't.

File: gpdb-doc/dita/admin_guide/workload_mgmt_resgroups.xml [hunk 8]
```diff
@@ -0,0 +1,375 @@
...
+    <body>
+      <p>When you install Greenplum Database, workload management using resource queues is enabled by default. To enable resource group-based workload management in Greenplum Database, you must set the <codeph><xref href="../ref_guide/config_params/guc-list.xml#gp_resource_manager" type="section"/></codeph> server configuration parameter.</p>
+      <ol id="ol_ec5_4dy_wq">
+        <li>Set the <codeph>gp_resource_manager</codeph> server configuration parameter to the value <codeph>"group"</codeph>:
+          <codeblock>gpconfig -s gp_resource_manager
+gpconfig -c gp_resource_manager -v "group"
+</codeblock>
+        </li>
+        <li>Restart Greenplum Database:
+            <codeblock>gpstart
```
Comment (gaos1) [inline gpdb-doc/dita/admin_guide/workload_mgmt_resgroups.xml]: restarting needs gpstop & gpstart

File: gpdb-doc/dita/admin_guide/workload_mgmt_resgroups.xml [hunk 9]
```diff
@@ -0,0 +1,375 @@
...
+          <codeblock>=# SELECT * FROM gp_toolkit.gp_resgroup_config;
+</codeblock>
+        </p>
+      </body>
+    </topic>
+
+    <topic id="topic23" xml:lang="en">
+      <title id="iz152239">Viewing Resource Group Query Status and CPU/Memory Usage</title>
+      <body>
+        <p>The <codeph><xref href="../ref_guide/system_catalogs/gp_resgroup_status.xml" type="topic" format="dita"/></codeph> <codeph>gp_toolkit</codeph> system view enables you to view the status and activity of a resource group. The view displays the number of running and queued transactions. It also displays the real-time CPU and memory usage of the resource group, averaged over all segments. To view this information:</p>
```
Comment (gaos1) [inline gpdb-doc/dita/admin_guide/workload_mgmt_resgroups.xml]: suggest removing "averaged over all segments". we will shown accurate usage on each segment.

File: gpdb-doc/dita/admin_guide/workload_mgmt_resgroups.xml [hunk 10]
```diff
@@ -0,0 +1,375 @@
...
+        <p>
+          <codeblock>=# SELECT * FROM gp_toolkit.gp_resgroup_status;
+</codeblock>
+        </p>
+      </body>
+    </topic>
+
+    <topic id="topic25" xml:lang="en">
+      <title id="iz152239">Viewing the Resource Group Assigned to a Role</title>
```
Comment (gaos1) [inline gpdb-doc/dita/admin_guide/workload_mgmt_resgroups.xml]: suggest adding one more section to check pg_stat_activity for query running or pending on resource group name and queue duration.

File: gpdb-doc/dita/admin_guide/workload_mgmt_resgroups.xml [hunk 11]
```diff
@@ -0,0 +1,375 @@
...
+        </p>
+
+        <p>Sample partial query output:</p>
+        <codeblock> rolname | rsgname  | procpid | waiting |     current_query     | datname 
+---------+----------+---------+---------+-----------------------+---------
+  sammy  | rg_light |  31861  |    f    | &lt;IDLE&gt; in transaction | testdb
+  billy  | rg_light |  31905  |    t    | SELECT * FROM topten; | testdb</codeblock>
+        <p>Use this output to identify the process id (<codeph>procpid</codeph>) of the transaction you want to cancel, and then cancel the process. For example, to cancel the pending query identified in the sample output above:</p>
+        <p>
+          <codeblock>=# pg_cancel_backend(31905) </codeblock>
```
Comment (gaos1) [inline gpdb-doc/dita/admin_guide/workload_mgmt_resgroups.xml]: should be "select pg_cancel_backend"

**Comment by lisakowen** [comment]
thank you for reviewing, @gaos1 .  i have incorporated your requested edits.

**Pull request merged**